src/volatile_memory.rs

```diff
@@ -119,6 +120,9 @@ pub fn compute_offset(base: usize, offset: usize) -> Result<usize> {
 
 /// Types that support raw volatile access to their data.
 pub trait VolatileMemory {
+    /// Type used for dirty memory tracking.
+    type B: Bitmap;
```

Comment:
This change will break too much existing code:(
And associated type defaults are unstable yet, so how about changing to
```
pub trait VolatileMemory<B: Bitmap = ()>
```